lib/stemcell/launcher.rb

```diff
@@ -80,6 +80,7 @@ def initialize(opts={})
       @aws_secret_key = opts['aws_secret_key']
       @aws_session_token = opts['aws_session_token']
       @max_attempts = opts['max_attempts'] || 3
+      @stub_responses = opts['stub_responses'] || false
```

Comment:
Shouldn't be needed as the `ec2` method should be intercepted tests.

Reply:
Doing that would result in a fruitless test since the ec2 client being initialized with the correct endpoint is what's being tested. Could remove stub_responses _and_ the test.

In SDK v1 endpoint was a global opt in Aws.config -- in SDK v3 it's a client option only as far as I could find.

Reply:
The config option cannot be tested without stubbing responses? I would expect checking the value of the endpoint config wouldn't require a remote operation.

Reply:
Creating a new client in v3 validates the endpoint config (by attempting to connect to it) if you don't enable stubbed responses.

Side note: the reason I'm setting it in the client and not AWS.config (global) is because other users of the SDK may be present in the same application.

Reply:
Can you add a reference here to where the endpoint validation is happening?

Reply:
I suspect it might have been requests to the EC2 metadata service (at 169.254.169.254), which should be fine since it's a link-local address.

Reply:
So, here's what I've observed:

If you run with AWS credentials available to a credential provider (such as `AWS_PROFILE` env var) then creating a new client will attempt to validate the endpoint resulting in an error (and test failure) such as:

```
2) Stemcell::Launcher#ec2 can return a client with custom endpoint
     Failure/Error: Aws::EC2::Client.new(opts)

     Aws::Errors::NoSuchEndpointError:
       Encountered a `SocketError` while attempting to connect to:

         https://sts.region1.amazonaws.com

       This is typically the result of an invalid `:region` option or a
       poorly formatted `:endpoint` option.

       * Avoid configuring the `:endpoint` option directly. Endpoints are constructed
         from the `:region`. The `:endpoint` option is reserved for certain services
         or for connecting to non-standard test endpoints.

       * Not every service is available in every region.

       * Never suffix region names with availability zones.
         Use "us-east-1", not "us-east-1a"

       Known AWS regions include (not specific to this service):

       af-south-1
...
```

This is what originally prompted me to add the stubbing option as I didn't want any network requests happening in unit tests. I verified that it does not try to validate even if the instance metadata api is reachable on an EC2 instance. That means that it's mostly a nuisance when developing locally if you've got valid credentials available since you have to unset them to run tests w/o `stub_responses`.